### PR TITLE
setup

### PR DESCRIPTION
**Pull request by KalebJohnson**
Basic build .env vars in readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,7 @@
-# backend
\ No newline at end of file
+# backend
+
+# JWTSIG
+SECRET="buildWeek"
+
+# PORT 
+PORT="5000"
```
Comment (wSedlacek) [inline README.md:7]: Can we turn this into a code block using ```?
That way it is very easy to copy and paste into a new file without needing to worry about syntax errors.

File: users-auth/auth-router.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	} catch(err) {
+		next(err)
+	}
+});
+
+
+router.post('/login', async (req, res, next) => {
+	try {
+		const { username, password, id } = req.body
+		const user = await Users.findBy({ username }).first()
```
Comment (wSedlacek) [inline users-auth/auth-router.js:34]: Username may not exist in the body and might throw from knex an error if undefined.

File: users-auth/auth-router.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		
+		if (!user) {
+			return res.status(401).json({
+				message: "Invalid Credentials",
+			})
+		}
+
+
+		const passwordValid = await bcrypt.compare(password, user.password)
```
Comment (wSedlacek) [inline users-auth/auth-router.js:43]: Password should be validated if it exists prior to using it.

File: database/migrations/20200916191917_users.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+exports.up = async function (knex) {
+    await knex.schema
+    .createTable("users", (tbl) => {
+        tbl.increments("id").unsigned().primary()
+        tbl.text("username", 20).unique().notNullable()
+        tbl.text("password", 128).notNullable();
```
Comment (wSedlacek) [inline database/migrations/20200916191917_users.js:6]: If these length limits are here can we check in the `add()` or `/login` if these constraints are met and send back useful error to the user.

File: database/config.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+const knex = require("knex")
+const knexfile = require("../knexfile")
+
+module.exports = knex(knexfile.development)
```
Comment (wSedlacek) [inline database/config.js:4]: In a future PR this will need to change to be dynamic based on the `NODE_ENV` or `DB_ENV`

**Comment by c00kamunga** [comment]
c00kamunga

**Review by c00kamunga — APPROVED** [review]

**Pull request merged**